packages/basic-crawler/src/internals/basic-crawler.ts

```diff
@@ -990,17 +993,55 @@ export class BasicCrawler<Context extends CrawlingContext = BasicCrawlingContext
     }
 
     protected async _handleFailedRequestHandler(crawlingContext: Context, error: Error): Promise<void> {
-        if (this.failedRequestHandler) {
-            await this._tagUserHandlerError(() => this.failedRequestHandler?.(crawlingContext, error));
-            return;
-        }
-
+        // Always log the last error regardless if the user provided a failedRequestHandler
         const { id, url, method, uniqueKey } = crawlingContext.request;
-        const message = error instanceof TimeoutError && !process.env.CRAWLEE_VERBOSE_LOG ? error.message : error.stack;
+        const message = this._getMessageFromError(error, true);
+
         this.log.error(
             `Request failed and reached maximum retries. ${message}`,
             { id, url, method, uniqueKey },
         );
+
+        if (this.failedRequestHandler) {
+            await this._tagUserHandlerError(() => this.failedRequestHandler?.(this._augmentContextWithDeprecatedError(crawlingContext, error), error));
+        }
+    }
+
+    /**
+     * Resolves the most verbose error message from a thrown error
+     * @param error The error received
+     * @returns The message to be logged
+     */
+    protected _getMessageFromError(error: Error, forceStack = false) {
+        return process.env.CRAWLEE_VERBOSE_LOG || forceStack ? error.stack ?? error.message ?? error : error;
```

Comment:
lets do at least this

```suggestion
        return (process.env.CRAWLEE_VERBOSE_LOG || forceStack)
            ? error.stack ?? error.message ?? error 
            : error;
```

Reply:
Good catch for needing () in the return branch! 😅 completely missed that